src/lib/io/sol-ipm.c

```diff
@@ -0,0 +1,370 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <errno.h>
+#include <inttypes.h>
+#include <stdlib.h>
+
+#ifndef SOL_LOG_DOMAIN
+#define SOL_LOG_DOMAIN &_sol_ipm_log_domain
+#endif
+
+#include "sol-log-internal.h"
+
+#include "ipm.h"
+#include "ipm/ipm_quark_se.h"
```

Comment:
Maybe, but couldn't we leave this to when it's needed? Them some common code could be factored out to some `sol-ipm-common.c`.